libs/s25main/ingameWindows/iwBuildings.cpp

```diff
@@ -52,24 +71,29 @@ iwBuildings::iwBuildings(GameWorldView& gwv, GameCommandFactory& gcFactory)
                    LOADER.GetImageN("resource", 41)),
       gwv(gwv), gcFactory(gcFactory)
 {
+    setBuildingOrder();
+    const unsigned rowCount = helpers::divCeil(bts.size(), 4);
+    Resize(iconSpacing * Extent(4, rowCount + 1) + bldContentOffset);
+
     const Nation playerNation = gwv.GetViewer().GetPlayer().nation;
     // Symbole für die einzelnen Gebäude erstellen
-    for(unsigned y = 0; y < bts.size() / 4 + (bts.size() % 4 > 0 ? 1 : 0); ++y)
+    for(unsigned y = 0; y < rowCount; y++)
     {
-        for(unsigned x = 0; x < 4; ++x)
+        for(unsigned x = 0; x < 4; x++)
```

Comment:
Please don't do such changes (no need to revert though either)
- There is no benefit in this case but it adds to the diff to review
- having a habbit of preincrement avoids thinking about the type
- `for(const x: helpers::range(4)` would be clearer if you want to improve such loops while changing them